wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceManager.java

```diff
@@ -695,6 +729,30 @@ private void requiredNotNull(Object object, String message) throws BadRequestExc
         }
     }
 
+    /*
+    * Get workspace using composite key.
+    *
+    */
+    private UsersWorkspaceImpl getByKey(String key) throws NotFoundException, ServerException {
+        String[] parts = key.split(":", -1); // -1 is to prevent skipping trailing part
+        switch (parts.length) {
+            case 2: {
+                String userName = parts[0];
+                String wsName = parts[1];
+                String ownerId;
+                if (userName.isEmpty()) {
+                    ownerId = getCurrentUserId();
```

Comment:
I am not sure that  current user's id is the right choice when username is empty. I think you should perform dao request with an empty username + i think that you can simplify this code by avoiding switch statement e.g:
```java
private UsersWorkspaceImpl getByKey(String key) throws NotFoundException, ServerException {
    String[] parts = key.split(":", -1); // -1 is to prevent skipping trailing part
    if (parts.length == 1) {
        return workspaceDao.get(key);
    }
    final String userName = parts[0];
    final String wsName = parts[1];
    final String ownerId = userManager.getByName(userName).getId();
    return workspaceDao.get(wsName, ownerId);
}
```